### PR TITLE
feat: wire up thorchain savers metadata/overview

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

First wire-up of Thorchain Savers metadata with functional overview modal.

Note, cap will need to bring a new concept of `opportunitySpecific` discriminated union a la `chainSpecific` - to be done in https://github.com/shapeshift/web/issues/3595.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/3584

## Risk

Low given Thorchain savers metadata resolution is under flag, though the selectors logic has been slightly revamped, which could cause regressions.

## Testing

- Enable the `SaversVaults` flag
- Ensure opportunity details is matching the one in https://app.thorswap.finance/earn / https://app.thoryield.com/savers

### Engineering

- This reads as sane code
- ☝🏽 

### Operations

-☝🏽doesn't need to be tested since this is under flag
- Clear your cache and ensure opportunities still load as before (note intermittent node failures can happen in all envs and are *not* related to this PR)

## Screenshots (if applicable)

image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3585** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3585" target="_blank">Graphite</a> 
    * **PR #3586** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3586" target="_blank">Graphite</a> 
      * **PR #3583** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3583" target="_blank">Graphite</a> 
        * **PR #3587** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3587" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3587?utm_source=stack-comment).

File: src/components/StakingVaults/AllEarnOpportunities.tsx [hunk 1]
```diff
@@ -63,8 +37,6 @@ export const AllEarnOpportunities = () => {
     dispatch,
   } = useWallet()
 
-  const { SaversVaults } = useSelector(selectFeatureFlags)
```
Comment (gomesalexandre) [inline src/components/StakingVaults/AllEarnOpportunities.tsx:66]: This now lives at resolver-level, the whole `saversVault` thing was a hack for the view-layer work to be done in isolation

File: src/state/slices/opportunitiesSlice/selectors.ts [hunk 2]
```diff
@@ -296,12 +296,14 @@ export const selectAggregatedEarnUserStakingOpportunities = createDeepEqualOutpu
 
       return Object.assign(
         {},
-        {
-          // TODO: The guts of getting contractAddress for Idle
-          // ETH/FOX opportunities contractAddress will be overwritten by STAKING_EARN_OPPORTUNITIES
-          // Can we generalize this? This is getting messy
-          contractAddress: fromAssetId(opportunity.underlyingAssetId).assetReference,
-        },
+        isToken(fromAssetId(opportunity.underlyingAssetId).assetReference)
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors.ts:299]: If the current asset isn't a token, we don't want a `contractAddress` property

File: src/state/slices/opportunitiesSlice/selectors.ts [hunk 3]
```diff
@@ -367,7 +371,7 @@ export const selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty =
       // Keep only the version with actual data if it exists, else keep the zero'd out version
       const aggregatedEarnUserStakingOpportunitiesIncludeEmpty = uniqBy(
         [...aggregatedEarnUserStakingOpportunities, ...emptyEarnOpportunitiesTypes],
-        'contractAddress',
+        ({ contractAddress, assetId }) => contractAddress ?? assetId,
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors.ts:374]: Leveraging the identity nature of `uniqBy` iteratee to either keep the uniques by `contractAddress` (e.g FOX Farming contracts) or by `assetId` for native assets

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx [hunk 4]
```diff
@@ -139,18 +111,33 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
...
   const opportunityData = useAppSelector(state =>
     selectEarnUserStakingOpportunityByUserStakingId(state, opportunityDataFilter),
   )
 
+  const underlyingAssetsFiatBalanceCryptoPrecision = useMemo(() => {
+    if (!asset || !opportunityData?.underlyingAssetId) return '0'
+
+    const cryptoAmount = bnOrZero(opportunityData?.stakedAmountCryptoBaseUnit)
+      .div(bn(10).pow(asset.precision))
+      .toString()
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:131]:
```suggestion
      .toFixed()
```

not sure if we want fixed here or string?

File: src/components/StakingVaults/AllEarnOpportunities.tsx (hunk 1 above)
Comment (0xdef1cafe) [reply src/components/StakingVaults/AllEarnOpportunities.tsx:66]: tiny issue - if we turn off flag the cached savers opportunities still appear, but this is a ways away from being enabled in prod so i'm not concerned with it for this PR

**Comment by 0xdef1cafe** [comment]
@gomesalexandre provisional stampy, LGTM

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx (hunk 4 above)
Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:131]: Copied from `FoxFarmingManager`, if that's an issue, let's update all occurrences in a follow-up PR @0xdef1cafe ?
Given all we're doing here is using this amount to calculate a rough fiat amount, this is safe IMO

**Review by 0xdef1cafe — APPROVED** [review]
stomped

**Pull request merged**